Assets/StateMachine/New Model/StateManager.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class StateManager : MonoBehaviour
+{
+    private BaseState CurrentState;
+    public OnStartState StartState = new OnStartState();
+    public OnGamePlayState GamePlay = new OnGamePlayState();
+    public OnPauseState PauseState = new OnPauseState();
+    public OnWinState WinState = new OnWinState();
+    public OnLoseState LoseState = new OnLoseState();
```

Comment:
we do not need these much classes, just initialize all the `State` s in awake and define their behaviour